.circleci/initialize.py

```diff
@@ -0,0 +1,392 @@
+    print("Initializing review app...")
+    apps = result["apps"]["apps"]
+    apps_name = result["apps"]["apps"][0]["name"]
+    # apps_owner = result["apps"]["apps"][0]["owner"]["username"]
+    # apps_status = result["apps"]["apps"][0]["status"]["running"]
+    # current_isAdmin = result["current"]["isAdmin"]
+    # apps_collaborators = result["apps"]["apps"][0]["collaborators"]
+    apps_permissionLevels = result["apps"]["apps"][0]["metadata"]
+    apps_linkedServices = result["apps"]["apps"][0]["linkedServices"]
+    apps_mounts = result["apps"]["apps"][0]["mounts"]
+    apps_environmentVariables = result["apps"]["apps"][0]["environmentVariables"]
+
+    permissionLevels = apps_permissionLevels
+    linkedServices = zip_list_index(apps_linkedServices, "serviceType", "name")
+    mounts = zip_list_index(apps_mounts, "hostDir", "targetDir")
+    environmentVariables = zip_list_index(apps_environmentVariables, "name", "value")
+
+    query = gql(
+        """
+        mutation (
+            $appname: String
+        ) {
+            addApp(
+                name: $appname
+            ) {
+                error
+            }
+        }
+        """
+    )
+
+    params = {"appname": APPNAME}
+
+    result = client_user.execute(query, variable_values=params)
+
+    print("  {APPNAME}".format(APPNAME=APPNAME))
+else:
+    print(result)
+    raise Exception(
+        "\nReview app not initialized"
+        + "\nApp {TARGET_APPNAME} does not exist or the user {USERNAME}".format(
+            TARGET_APPNAME=TARGET_APPNAME, USERNAME=USERNAME
+        )
+        + "does not have permission to query this app.\n"
+    )
+
+if len(linkedServices.items()) != 0:
+    print(
+        "Adding redis/postgres databases...",
+    )
+    for serviceType in linkedServices:
+        if serviceType == "redis":
+            serviceName = "{APPNAME}-{serviceType}".format(
+                APPNAME=APPNAME,
+                serviceType="redis",
+            )[0:30]
+            query_addService = gql(
+                """
+                mutation (
+                    $serviceType: ServiceType=redis,
+                    $serviceName: String
+                ) {
+                    addService (
+                        name: $serviceName,
+                        serviceType: $serviceType
+                    ) {
+                        error
+                    }
+                }
+                """
+            )
+            params_addService = {
+                "serviceName": serviceName,
+                "serviceType": serviceType,
+            }
+            sleep(5)
+            result = client_service.execute(
+                query_addService, variable_values=params_addService
+            )
+
+            print(
+                "  {serviceName}, {serviceType}".format(
+                    serviceName=serviceName, serviceType=serviceType
+                )
+            )
+        elif serviceType == "postgres":
```

Comment:
🔪 this block once the top block is more DRY. Then dedent the top block as there won't be any more `if` statements